### ISSUE TITLE
Array formulas should either be classed as formula & array type or not.

### ISSUE THREAD

**Issue by MartinDawson** [handsontable/hyperformula#781]
### Description
There's a mismatch in functions for the cells with an ARRAY type that are also formulas.

See here:

![image](https://user-images.githubusercontent.com/15030491/126896049-c0652540-09e8-4c6f-8135-fe365a2dc517.png)

`doesCellHaveFormula()` returns false.
But `getSheetFormulas()` returns it as part of it's results.

This is mis-leading to the user. It should either be classes as a formula or not but not in between.

### Steps to reproduce
```js
const sheetName = hf.addSheet(state.currentSheet);
const sheetId = hf.getSheetId(sheetName);

// Fill the HyperFormula sheet with data.
hf.setSheetContent(sheetId, [[0, 1]]);
hf.setCellContents(
  { sheet: sheetId, row: 0, col: 2 },
  "=ARRAYFORMULA(ISEVEN(A1:A2*10))"
);

console.log(hf.doesCellHaveFormula({ sheet: sheetId, row: 0, col: 2 }));
console.log(hf.getSheetFormulas(sheetId));
```

### Demo
https://codesandbox.io/s/fervent-cannon-voj87?file=/src/hyperformulaConfig.js

### Your environment

* HyperFormula version: 1.0.0
* Browser Name and version: Chrome latest
* Operating System: Windows 10

**Comment by izulin** [comment]
Fixed in #790

**Comment by wojciechczerniak** [comment]
Thanks @izulin !
